### PR TITLE
feat: [gd] invert_hash readeable model with hexfloat

### PR DESCRIPTION
**Pull request by lalo**
This change is useful to hunt down bugs, floats might print the same (bc of rounding, etc) but if we print the hex repr we can identify exactly when do two VW/model instances diverge.

now:
```
-Action^fff*User^a:431:-0.0420591 3.99279 1
-Action^fff*_ccb_slot_index^index1:1661:-0.0545174 4.26212 1
-Slot^h*User^c:2034:-0.0509719 1.3541 1
-User^a*_ccb_slot_index^index2:2511:-0.0336862 2.16077 1
```
with --hexfloat_weights:
```
+Action^fff*User^a:431:-0x1.588c6p-5 0x1.ff13a2p+1 0x1p+0
+Action^fff*_ccb_slot_index^index1:1661:-0x1.be9b4p-5 0x1.10c68ep+2 0x1p+0
+Slot^h*User^c:2034:-0x1.a18fecp-5 0x1.5aa682p+0 0x1p+0
+User^a*_ccb_slot_index^index2:2511:-0x1.13f518p-5 0x1.14941ap+1 0x1p+0
```

### CODE REVIEW

File: .github/workflows/asan.yml [hunk 1]
```diff
@@ -52,8 +52,11 @@ jobs:
         run: |
           cd build
           ctest --verbose --output-on-failure
+      - name: Run python test script - win
```
Comment (olgavrou) [inline .github/workflows/asan.yml:55]: add a comment as to why

File: vowpalwabbit/core/src/reductions/gd.cc [hunk 2]
```diff
@@ -1022,6 +1022,8 @@ void save_load_online_state_weights(VW::workspace& all, io_buf& model_file, bool
   }
   else
   {  // write binary or text
+    if (all.hexfloat_weights && (text || all.print_invert)) { msg << std::hexfloat; }
```
Comment (olgavrou) [inline vowpalwabbit/core/src/reductions/gd.cc:1025]: do we need to un `std::hexfloat` here?

Comment (olgavrou) [reply vowpalwabbit/core/src/reductions/gd.cc:1025]: no nevermind

File: .github/workflows/asan.yml (hunk 1 above)
Comment (lalo) [reply .github/workflows/asan.yml:55]: done

**Review by olgavrou — APPROVED** [review]

**Pull request merged**